troubleshooting-exercise.md

```diff
@@ -0,0 +1,1126 @@
+           value  ------------- Distribution ------------- count    
+      2147483648 |                                         0        
+      4294967296 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@ 6807     
+      8589934592 |                                         0 
+```
+
+</details>
+
+#### Writing to persistent storage duration
+
+Given the new approach to correlate probes, the `persist-async-start` probe is no longer necessary. The `persist-start` probe can be used instead.
+
+<details>
+
+#### Running a single client:
+
+```console
+Summary of all writes to persistent storage taken in one minute represented in nanoseconds:
+  total writes                                                     60
+  average write                                              64737102
+  max write                                                  78047417
+  min write                                                  52074250
+  visualisation of writes                           
+           value  ------------- Distribution ------------- count    
+        16777216 |                                         0        
+        33554432 |@@@@@@@@@@@@@@@@@@@@@@@@@@@              40       
+        67108864 |@@@@@@@@@@@@@                            20       
+       134217728 |                                         0   
+```
+
+#### Under light load (15 clients):
+
+```console
+Summary of all writes to persistent storage taken in one minute represented in nanoseconds:
+  total writes                                                    885
+  average write                                              61787007
+  max write                                                  81290167
+  min write                                                  51105667
+  visualisation of writes                           
+           value  ------------- Distribution ------------- count    
+        16777216 |                                         0        
+        33554432 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@      773      
+        67108864 |@@@@@                                    112      
+       134217728 |                                         0 
+```
+
+#### Under medium load (125 clients):
+
+```console
+Summary of all writes to persistent storage taken in one minute represented in nanoseconds:
+  total writes                                                   7413
+  average write                                              61725998
+  max write                                                  79233667
+  min write                                                  50252000
+  visualisation of writes                           
+           value  ------------- Distribution ------------- count    
+        16777216 |                                         0        
+        33554432 |@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@       6354     
+        67108864 |@@@@@@                                   1059     
+       134217728 |                                         0 
```

Comment:
This looks much better!

The `quantize` aggregating function doesn't give us a huge amount of granularity. If you want to dissect that distribution a bit, you could look at the `lquantize` or `llquantize` functions. It might also be interesting to look at for requests that block vs. requests that do not block to see how each contributes to the total picture.

Reply:
Thanks for the tip! Will try those out :D